ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Init.hs

```diff
@@ -68,7 +69,12 @@ forkSyncStateOnTipPointChange registry menv =
         }
   where
     action :: Point blk -> m ()
-    action _tipPoint = void $ implSyncWithLedger menv
+    action _tipPoint = do
+      !start <- getMonotonicTime
+      void $ implSyncWithLedger menv
+      !end <- getMonotonicTime
+      traceWith (mpEnvTracer menv) $ TraceMempoolSynced
+          (diffTime end start)
```

Comment:
@jasagredo recently added a small combinator for this to `O.C.Util.Enclose`:
```suggestion
    action _tipPoint =
      encloseTimedWith (TraceMempoolSynced >$< mpEnvTracer menv) $
        void $ implSyncWithLedger menv
```
(also would require changing the payload of `TraceMempoolSynced` to `EnclosingTimed`)

Reply:
:+1: Nice.

Will make the change.